.github/workflows/build-publish-mcr.yml

```diff
@@ -3,47 +3,24 @@
 
 name: Build and publish to MCR
 on:
-  workflow_dispatch:
-    inputs:
-      releaseTag:
-        description: 'Release tag to publish, defaults to the latest one'
-        type: string
+  push:
+    # Only release on supported semantic version tagging e.g. v0.0.1-rc.0
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-rc.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+'
```

Comment:
i like the enforcement based on tags in this format. Means we can never accidentally screw up our versioning due to alha typo or something like that.

Reply:
Agreed.